parsebib.el

```diff
@@ -240,113 +240,144 @@ trailing space will be included in the overall match."
+be strings.
+
+The car of this cons cell is the COMMAND,capturing exactly one
+obligatory argument. The argument is concatenated with the accent
+to produce the display string.")
+
+(defvar parsebib-TeX-literal-replacement-alist
+  ;; LaTeX2 Escapable "Special" Characters
+  `(("\\%" . "%") ("\\&" . "&") ("\\#" . "#") ("\\$" . "$")
     ;; Quotes
     ("``" . "\N{LEFT DOUBLE QUOTATION MARK}")
     ("`"  . "\N{LEFT SINGLE QUOTATION MARK}")
     ("''" . "\N{RIGHT DOUBLE QUOTATION MARK}")
     ("'"  . "\N{RIGHT SINGLE QUOTATION MARK}")
+    ;; Dashes
+    ("---" . "\N{EM DASH}")
+    ("--" . "\N{EN DASH}")
+    ;; Remove all remaining {braces}
+    ("{" . "") ("}" . ""))
+  "Alist of LITERAL-REPLACEMENT pairs. Both are strings.
 
-    ;; Formatting Commands
-    ("\\\\textit{\\(.*?\\)}" . parsebib--convert-tex-italics)
-    ("\\\\emph{\\(.*?\\)}"   . parsebib--convert-tex-italics)
-    ("\\\\textbf{\\(.*?\\)}" . parsebib--convert-tex-bold)
-    ("\\\\textsc{\\(.*?\\)}" . parsebib--convert-tex-small-caps)
+Note that adding such a pair to this variable has no effect
+unless `parsebib-TeX-markup-replacement-alist' is adjusted
+accordingly. For example, after adding LITERAL-REPLACEMNT
+calling,
 
-    ;; Non-ASCII Letters (Excluding Accented Letters)
-    ,@(mapcar
-       (apply-partially 'apply 'parsebib--build-TeX-command-regexp)
-       '(("AA" "\N{LATIN CAPITAL LETTER A WITH RING ABOVE}")
-         ("AE" "\N{LATIN CAPITAL LETTER AE}")
-         ("DH" "\N{LATIN CAPITAL LETTER ETH}")
-         ("DJ" "\N{LATIN CAPITAL LETTER ETH}")
-         ("L"  "\N{LATIN CAPITAL LETTER L WITH STROKE}")
-         ("SS" "\N{LATIN CAPITAL LETTER SHARP S}")
-         ("NG" "\N{LATIN CAPITAL LETTER ENG}")
-         ("OE" "\N{LATIN CAPITAL LIGATURE OE}")
-         ("O"  "\N{LATIN CAPITAL LETTER O WITH STROKE}")
-         ("TH" "\N{LATIN CAPITAL LETTER THORN}")
-
-         ("aa" "\N{LATIN SMALL LETTER A WITH RING ABOVE}")
-         ("ae" "\N{LATIN SMALL LETTER AE}")
-         ("dh" "\N{LATIN SMALL LETTER ETH}")
-         ("dj" "\N{LATIN SMALL LETTER ETH}")
-         ("l"  "\N{LATIN SMALL LETTER L WITH STROKE}")
-         ("ss" "\N{LATIN SMALL LETTER SHARP S}")
-         ("ng" "\N{LATIN SMALL LETTER ENG}")
-         ("oe" "\N{LATIN SMALL LIGATURE OE}")
-         ("o"  "\N{LATIN SMALL LETTER O WITH STROKE}")
-         ("th" "\N{LATIN SMALL LETTER THORN}")
-
-         ("ij" "ij")
-         ("i"  "\N{LATIN SMALL LETTER DOTLESS I}")
-         ("j"  "\N{LATIN SMALL LETTER DOTLESS J}")))
-
-    ;; Commands with obligatory non-empty argument
-    ("\\\\[a-zA-Z*]+\\(?:\\[.*\\]\\)?{\\(.+?\\)}" . "\\1")
-
-    ;; Commands without arguments, optionally terminated by empty braces
-    ("\\(\\\\[a-zA-Z*]+\\)\\(?:\\[.*\\]\\)?\\(?:{}\\)?" . "\\1")
-
-    ;; Collapse white space
-    ("[[:blank:]]+" . " ")
+`(cl-callf (lambda (regex) (rx (or LITERAL (regexp regex))))
+    (alist-get (quote parsebib--replace-literal)
+               parsebib-TeX-markup-replacement-alist))'
+
```

Comment:
I'd prefer two spaces before the code snippet and no `...':

```
  (cl-callf (lambda (regex) (rx (or LITERAL (regexp regex))))
      (alist-get (quote parsebib--replace-literal)
                 parsebib-TeX-markup-replacement-alist))
```

Reply:
Done